src/puppetlabs/trapperkeeper/services/webserver/jetty9_config.clj

```diff
@@ -323,14 +323,18 @@
              "Error: More than one default server specified in configuration"))))
 
 (defn selectors-count
-  "The number of selector threads that should be allocated per connector per core."
+  "The number of selector threads that should be allocated per connector per
+  core.  This algorithm duplicates the default that Jetty 9.2.7.v20150116 uses.
+  See: https://github.com/eclipse/jetty.project/blob/jetty-9.2.7.v20150116/jetty-server/src/main/java/org/eclipse/jetty/server/ServerConnector.java#L229"
   [num-cpus]
-  num-cpus)
+  (max 1 (min 4 (int (/ num-cpus 2)))))
```

Comment:
Yeah, the defaults that Jetty uses internally did change between 9.1.0 and 9.2.7.  Our current implementation on tk-jetty9 master matches the Jetty 9.1.0 defaults.  This PR would make the tk-jetty9 implementation mirror the Jetty 9.2.7 implementation for these defaults.  Arguably, it would be better instead to keep the defaults the same in the tk-jetty9 service for compatibility.  I talked with @cprice404 about this a bit and we were thinking it might be better to just follow core Jetty's lead since the maintainers undoubtedly have more insight than we do on how to tune these.

It may make sense sometime down the road to provide a way for selectors and acceptors to be configured externally, i.e., through the webserver configuration.  At that point, honoring the values set in the configuration file across releases would be a must but, in the absence of that, these values are more of an internal configuration detail and it may be best for now to follow core Jetty's lead for the auto-tuning.